src/main/java/com/novemberain/quartz/mongodb/db/MongoConnector.java

```diff
@@ -110,10 +130,56 @@ private MongoClient createClient() throws SchedulerConfigException {
             }
         }
 
-        private MongoClientOptions createOptions() {
+        private MongoClientOptions createOptions() throws SchedulerConfigException {
+            SSLContext sslContext = getSSLContext();
+            if (sslContext == null) {
+                if (enableSSL != null) {
+                    optionsBuilder.sslEnabled(enableSSL);
+                    if (sslInvalidHostNameAllowed != null) {
+                        optionsBuilder.sslInvalidHostNameAllowed(sslInvalidHostNameAllowed);
+                    }
+                }
+            } else {
+                optionsBuilder.sslEnabled(true);
+                optionsBuilder.sslContext(sslContext);
+            }
             return optionsBuilder.build();
         }
 
+        private SSLContext getSSLContext() throws SchedulerConfigException {
+            try {
+                KeyStore trustStore = loadKeyStore(trustStorePath, trustStorePassword, trustStoreType);
+                KeyStore keyStore = loadKeyStore(keyStorePath, keyStorePassword, keyStoreType);
+                if (trustStore == null && keyStore == null) {
+                    return null;
+                }
+                SSLContextBuilder sslContextBuilder = SSLContexts.custom();
+                if (trustStore != null) {
+                    sslContextBuilder.loadTrustMaterial(trustStore, null);
```

Comment:
Same comment as in #2, why no password?

Reply:
See https://github.com/nuxeo/quartz-mongodb/pull/2#discussion_r330380267.